src/BPINN_ode.jl

```diff
@@ -99,48 +99,33 @@ struct BNNODE{C, K, ST <: Union{Nothing, AbstractTrainingStrategy}, IT, A, M,
     l2std::Vector{Float64}
     phystd::Vector{Float64}
     dataset::D
-    init_params::I
     physdt::Float64
+    MCMCargs::H
     nchains::Int64
-    autodiff::Bool
+    init_params::I
     Integrator::IT
     Adaptor::A
-    targetacceptancerate::Float64
     Metric::M
+    targetacceptancerate::Float64
     jitter_rate::Float64
     tempering_rate::Float64
-    max_depth::Int64
-    Δ_max::Int64
-    n_leapfrog::Int64
-    δ::Float64
-    λ::Float64
+    autodiff::Bool
     progress::Bool
     verbose::Bool
-
-    function BNNODE(chain, Kernel = HMC; strategy = nothing,
-        draw_samples = 2000,
-        priorsNNw = (0.0, 2.0), param = [nothing], l2std = [0.05],
-        phystd = [0.05], dataset = [nothing],
-        init_params = nothing,
-        physdt = 1 / 20.0, nchains = 1,
-        autodiff = false, Integrator = Leapfrog,
-        Adaptor = StanHMCAdaptor, targetacceptancerate = 0.8,
-        Metric = DiagEuclideanMetric, jitter_rate = 3.0,
-        tempering_rate = 3.0, max_depth = 10, Δ_max = 1000,
-        n_leapfrog = 20, δ = 0.65, λ = 0.3, progress = false,
-        verbose = false)
-        new{typeof(chain), typeof(Kernel), typeof(strategy), typeof(Integrator),
-            typeof(Adaptor),
-            typeof(Metric), typeof(init_params), typeof(param),
-            typeof(dataset)}(chain, Kernel, strategy, draw_samples,
-            priorsNNw, param, l2std,
-            phystd, dataset, init_params,
-            physdt, nchains, autodiff, Integrator,
-            Adaptor, targetacceptancerate,
-            Metric, jitter_rate, tempering_rate,
-            max_depth, Δ_max, n_leapfrog,
-            δ, λ, progress, verbose)
-    end
+end
+function BNNODE(chain, Kernel = HMC; strategy = nothing, draw_samples = 2000,
+    priorsNNw = (0.0, 2.0), param = nothing, l2std = [0.05], phystd = [0.05],
+    dataset = [nothing], physdt = 1 / 20.0, MCMCargs = (n_leapfrog = 30), nchains = 1,
```

Comment:
This is not a namedtuple, also why not put everything (Integrator, Adaptor, Metric, target acceptance etc) into this? It should be documented in the docstring but the arguments can be reduced here. Also make the name MCMCkwargs instead of args since we expect a namedtuple and they are all kwargs to the downstream functions

Reply:
in the case of NUTS and HMCDA this becomes a NamedTuple, ive let this become an Int in the case of HMC as the format of input(MCMCargs) is similar to case of NUTS and HMCDA for the user. For making this a NamedTuple I would have had to make it MCMCargs = (n_leapfrog=30,)  instead of MCMCargs = (n_leapfrog=30) ,that extra comma can be unintuitive for case of HMC.

Reply:
>(n_leapfrog=30,) instead of MCMCargs = (n_leapfrog=30) ,that extra comma can be unintuitive for case of HMC.

I think most of our users are usually familiar with namedtuple so that should be fine, it makes sense to keep the type the same across the sampling algs.

>Also should i put integrator/Adaptor related args in seperate Nametuples(as they are seperate entities from the sampling alg)?

Yeah, that should be fine as well. If the fields you need to pass are not going to vary too much between the methods make it a struct.